src/Spaces/quadrature.jl

```diff
@@ -42,41 +42,135 @@ function quadrature_points end
 
 
 """
-    GLL{Nq}()
+    LegendreGaussLobatto
+    LGL
 
-Gauss-Legendre-Lobatto quadrature using `Nq` quadrature points.
+A Legendre-Gauss-Lobatto (LGL) quadrature rule.
+
+# Constructors
+
+    LegendreGaussLobatto(npoints=N)
+    LGL(npoints=N)
+
+Construct the quadrature rule with `N` quadrature points (`N ≥ 2`).
+
+    LegendreGaussLobatto(degree=p)
+    LGL(degree=p)
+
+Construct the quadrature rule with polynomial degree `p` (i.e. `p+1` quadrature
+points, with `p ≥ 1`).
```

Comment:
Thanks Simon. I think this docstring can lead to some confusion. If we are talking about constructing a _basis_ such as an H1 Lagrange basis, then I would agree with using a phrase  such as `Construct a finite element H1 Lagrange basis with polynomial degree `p` (i.e. `p+1` nodes, with `p ≥ 1`).` 
But a quadrature rule generally can have an arbitrary number of quadrature points, so I would use:
```suggestion
    LegendreGaussLobatto(qpoints=q)
    LGL(qpoints=q)
    
Construct the quadrature rule with quadrature points `q` (i.e. `q` quadrature
points, with `q` arbitrary).
```
`q` could be totally independent of `p`. It could be equal to `p`, it could be less than `p` for cases of underintegration and can be more than `p` for cases of overintegration. That's why is useful expressing these two concepts that are distinct with different variable names.

Reply:
I'm not sure I understand what you're proposing here: what's the difference between `npoints` and `qpoints`?

Reply:
> `q` could be totally independent of `p`. It could be equal to `p`, it could be less than `p` for cases of underintegration and can be more than `p` for cases of overintegration.

At the moment, that requires constructing two spaces with two different quadrature rules (one for the storage space, one for the integration space). 

The purpose of this PR to allow specifying either the number of points, or the degree of the polynomial that you can approximate with those points (which I thought was the point you were raising in #163).

Reply:
> Fixes #163. Renames the quadarture rules (and aliases), and adds keyword constructors so that you can specify either number of points, or polynomial degree.
> 
> Would be good to get some input on the exact names of (a) the quadrature rules, and (b) the keyword arguments. @valeriabarra @sriharshakandala any thoughts?

The `degree`/'npoints' keyword argument looks good to me.

Reply:
They are conceptually different things and `qpoints` can be arbitrary, while the number of nodes, if you want to call them `npoints`, are not. `npoints = p+1` always, by definition. E.g., for a quadratic basis, you have `npoints = p + 1 \equiv 3` nodes. But you can decide to have lower, equal, or higher order integration (generally overintegration is preferred for obvious reasons, not underintegration) so that at the end when you do integration across all elements, you can have a desired order for the integration of non-linear terms (e.g., `u^2` terms) that does not necessarily coincide with the order chosen to represent the state variable `u`

Reply:
@simonbyrne : One option we can think of is to evaluate the degree `d` polynomials on an user specified grid!